daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -673,7 +673,7 @@ sealed class ElementRuntimeData(
   def isSimpleType = optPrimType.isDefined
 
   def schemaURIStringsForFullValidation = schemaURIStringsForFullValidation1.distinct
-  private def schemaURIStringsForFullValidation1: Seq[String] = (schemaFileLocation.uriString +:
+  def schemaURIStringsForFullValidation1: Seq[String] = (schemaFileLocation.uriString +:
```

Comment:
opened this up for https://github.com/apache/incubator-daffodil/pull/431/files#diff-f6dc861db0417ffe6d0f80f7e0a166757345c5a3505e76de8f3bb694c1d170f3R259

Reply:
No I don't get it. Why isn't just using the public schemaURIStringForFullValidation sufficient? 

This idiom, a public member that just uses a private def that is recursive to compute it, is common.

btw: this public member should definitely be a lazy val, not a def. So please change that also. No need for this to be computed over and over.

Reply:
> No I don't get it. Why isn't just using the public schemaURIStringForFullValidation sufficient?

Good question.  I have no idea really, but was just trying to keep the original behavior as consistent as possible through my changes.  I havent saw any tests break while refactoring so Im thinking there is no coverage on these parts :)

I did make that package private in a commit that snuck in during your review.

Reply:
> btw: this public member should definitely be a lazy val, not a def. So please change that also. No need for this to be computed over and over.

done

Reply:
There we go.  I think its all in order now.